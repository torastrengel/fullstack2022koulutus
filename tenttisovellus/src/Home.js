const Home = () => {
  return (
    <div className="home-container">
      <img
        className="home-picture"
        src="/images/home.png"
        alt="abstract human standing"
      />
      <div className="home-introduction">
        <h1>Tervetuloa tenttisovellukseen! üçú</h1>
        <p>
          Tenttisovelluksen avulla sinusta tulee entist√§kin fiksumpi ihminen.
          Sovelluksen k√§yt√∂n j√§lkeen j√§√§t miettim√§√§n sit√§ faktaa, miten oletkaan
          p√§rj√§nnyt ilman t√§t√§ aiemmin.
        </p>
      </div>
    </div>
  );
};

export default Home;
