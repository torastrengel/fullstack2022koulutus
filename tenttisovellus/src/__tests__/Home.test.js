import Home from '../Home';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

it('Home, snapshot test', () => {
  const component = renderer.create(<Home />);
  let tree = component.toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});

test('Is hero image present in the Home component', () => {
  render(<Home />);
  const heroImage = screen.getByAltText('abstract human standing');
  expect(heroImage).toBeInTheDocument();
});

test('Is title present on the landing page', () => {
  render(<Home />);
  const heading = screen.getByText('Tervetuloa tenttisovellukseen! 🍜');
  expect(heading).toBeInTheDocument();
});
