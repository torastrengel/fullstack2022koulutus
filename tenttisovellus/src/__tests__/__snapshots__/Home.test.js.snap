// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home, snapshot test 1`] = `
<div
  className="home-container"
>
  <img
    alt="abstract human standing"
    className="home-picture"
    src="/images/home.png"
  />
  <div
    className="home-introduction"
  >
    <h1>
      Tervetuloa tenttisovellukseen! üçú
    </h1>
    <p>
      Tenttisovelluksen avulla sinusta tulee entist√§kin fiksumpi ihminen. Sovelluksen k√§yt√∂n j√§lkeen j√§√§t miettim√§√§n sit√§ faktaa, miten oletkaan p√§rj√§nnyt ilman t√§t√§ aiemmin.
    </p>
  </div>
</div>
`;
